### PR TITLE
Update release notes based on r13.0.0rc2 review

### PR DESCRIPTION
**Pull request by git-harry**
The release notes for r13.0.0rc2 were reviewed in preparation for the
release of r13.0.0. Of the changes to be agreed based on that review the
following are relevant to newer versions of the notes:

- a test note in the repo is removed
- a note about L3HA has been expanded to explain why it's being disabled
- the prelude sections have been removed because they weren't adding
  value
- references to VXLAN interfaces have been replaced with VXLAN networks
  to be more accurate
- a note regarding overlay networks has been reworded to better explain
  the change to which it refers
- security hardening note has been adjusted to keep to control interface
  consistent for all deployment paths
- the removal of LVMISCSIDriver by Cinder is noted

In addition to the changes in the cherry-pick, the upgrades sections of
the notes have been removed because upgrades are not supported in
r13.0.0.

Connected https://github.com/rcbops/u-suk-dev/issues/398

(cherry picked from commit 73daaf3393c0d6987e989ed8c01a4cf542122d44)

### CODE REVIEW

**Comment by evrardjp** [comment]
Missing (because of removals):

``other: `neutron_l2_population` is set to `True` in this release.``

File: releasenotes/notes/turn-off-l3ha-18274d19d54a6315.yaml [hunk 1]
```diff
@@ -1,20 +1,6 @@
...
-
-    For more information, see:
-    `bug 1149 <https://github.com/rcbops/rpc-openstack/issues/1149>`_
-    or the known issues section in the v13.0 RPCO documentation.
-fixes:
-  - Fixed `bug 1149 <https://github.com/rcbops/rpc-openstack/issues/1149>`_
 other:
-  - Neutron L3HA as a default has been disabled.
+  - Neutron L3HA as a default has been disabled. This has been done
+    to due to the number of bugs that still exist in that feature.
```
Comment (mancdaz) [inline releasenotes/notes/turn-off-l3ha-18274d19d54a6315.yaml]: s/to due to/due to/

**Comment by mancdaz** [comment]
one inline comment

File: releasenotes/notes/LVMISCSIDriver-b25cdd70ef0c5a4d.yaml [hunk 2]
```diff
@@ -0,0 +1,4 @@
+---
+other:
+ - The cinder driver LVMISCSIDriver has been removed, any configurations should be
+   updated to use LVMVolumeDriver instead prior to deploying Mitaka.
```
Comment (sigmavirus24) [inline releasenotes/notes/LVMISCSIDriver-b25cdd70ef0c5a4d.yaml]: 👍

**Comment by sigmavirus24** [comment]
👍

File: releasenotes/notes/override-l2pop-default-d7426c07d8b72f2e.yaml [hunk 3]
```diff
@@ -1,10 +1,4 @@
 ---
-issues:
-  - The default value for `neutron_l2_population` has been
-    overridden, which will require deployers to recreate
-    any VXLAN interfaces created with `neutron_l2_population`
-    set `False`. Not recreating these interfaces will lead to
-    connectivity issues for virtual machines.
-upgrade:
+other:
   - Based on feedback from Support, the default value for
```
Comment (evrardjp) [inline releasenotes/notes/override-l2pop-default-d7426c07d8b72f2e.yaml]: ``Based on feedback from our support team, ``

It sounds better if it's to be public.

**Comment by evrardjp** [comment]
See my (optional) comment.

👍

**Comment by asettle** [comment]
I'll address JP's comments in my own patch :)

**Comment by erikmwilson** [comment]
I'm taking this over for @asettle so let me know if there are any more changes.

**Pull request merged**